src/status_im2/contexts/chat/composer/link_preview/events.cljs

```diff
@@ -0,0 +1,132 @@
+(ns status-im2.contexts.chat.composer.link-preview.events
+  (:require
+    [clojure.set :as set]
+    [clojure.string :as string]
+    [status-im.data-store.messages :as data-store.messages]
+    [taoensso.timbre :as log]
+    [utils.collection]
+    [utils.re-frame :as rf]))
+
```

Comment:
@ilmotta I have just noticed that the events for link previews are implemented in a file named `events` next to `view` and `style`. Is this a guideline we have?  I could be doing it wrong, but I usually add the events in `status_im` namespace, for example `status-im.chat.models.mentions`

Reply:
Hi @OmarBasem. Events should live in their own files, in a directory "close" to the context or feature in question. https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md#re-frame-events I never commented on the events in `chat.models` because I thought the person would eventually migrate to the new structure.

The choice of which directory to store the events should be obvious most of the time. If it's not, then it probably means the events file could be split, or it could be moved up in the directory hierarchy, e.g. `status-im2.contexts.activity-center.events`. ns `status-im2.contexts.chat.messages.delete-message-for-me.events` is also a good example, which also has unit tests.

Reply:
@ilmotta So I guess we need to refactor the composer events and subscriptions to the new composer namespace

Reply:
That'd be great @OmarBasem. More things in `status-im2` and less in `status-im` :)